src/features/selectedPost/selectedPostSlice.ts

```diff
@@ -0,0 +1,29 @@
+/* eslint-disable no-param-reassign */
+import { createSlice, PayloadAction } from '@reduxjs/toolkit';
+import { Post } from '../../types/Post';
+
+type InitialState = {
+  selectedPost: Post | null;
+};
+
+const initialState: InitialState = {
+  selectedPost: null,
+};
+
+const saveSelectedPostSlice = createSlice({
```

Comment:
Consider renaming `saveSelectedPostSlice` to `selectedPostSlice` to better reflect its purpose of managing the selected post state, including both saving and clearing actions.